extension/basicauthextension/extension.go

```diff
@@ -177,3 +198,50 @@ func (a *authData) GetAttribute(name string) interface{} {
 func (*authData) GetAttributeNames() []string {
 	return []string{"username", "raw"}
 }
+
+type perRPCAuth struct {
+	metadata map[string]string
+}
+
+// GetRequestMetadata returns the request metadata to be used with the RPC.
+func (p *perRPCAuth) GetRequestMetadata(context.Context, ...string) (map[string]string, error) {
+	return p.metadata, nil
+}
+
+// RequireTransportSecurity always returns true for this implementation.
+func (p *perRPCAuth) RequireTransportSecurity() bool {
+	return true
+}
+
+type basicAuthRoundTripper struct {
+	base     http.RoundTripper
+	authData *ClientAuthSettings
+}
+
+func (b *basicAuthRoundTripper) RoundTrip(request *http.Request) (*http.Response, error) {
+	newRequest := request.Clone(request.Context())
+	newRequest.SetBasicAuth(b.authData.Username, b.authData.Password)
+	return b.base.RoundTrip(newRequest)
+}
+
+func (ba *basicAuth) RoundTripper(base http.RoundTripper) (http.RoundTripper, error) {
+	if strings.Contains(ba.clientAuth.Username, ":") {
+		return nil, errInvalidFormat
+	}
+	return &basicAuthRoundTripper{
+		base:     base,
+		authData: ba.clientAuth,
+	}, nil
+}
+
+func (ba *basicAuth) PerRPCCredentials() (creds.PerRPCCredentials, error) {
+	if strings.Contains(ba.clientAuth.Username, ":") {
```

Comment:
You probably don't need this check here, do you?

Reply:
Are you suggesting this check shouldn't be here and be somewhere else or not present at all?

Reply:
The constraint is only about HTTP headers, not sure whether the gRPC header would have the same limitation. Leave it here for the moment, I think it makes more sense to do it here than not have it checked at all.